dockerfiles/three_cards_divination.Dockerfile

```diff
@@ -5,13 +5,13 @@ USER spring:spring
 
 WORKDIR /app
 
-COPY ../.mvn .mvn
-COPY ../mvnw pom.xml ./
-COPY ../three-cards-divination/pom.xml ./three-cards-divination/
+COPY --chown=spring:spring .mvn .mvn
+COPY --chown=spring:spring mvnw pom.xml ./
+COPY --chown=spring:spring three-cards-divination/pom.xml ./three-cards-divination/
 RUN ./mvnw dependency:resolve
-COPY ../three-cards-divination/src ./three-cards-divination/src
-RUN ./mvnw package
-COPY ../three-cards-divination/target ./three-cards-divination/target
+COPY --chown=spring:spring three-cards-divination/src ./three-cards-divination/src
+RUN ./mvnw package -DskipTests
+COPY --chown=spring:spring three-cards-divination/target ./three-cards-divination/target
```

Comment:
**Suggestion:** Consider using a `.dockerignore` file to exclude files and directories that are not needed for building the Docker image. This can help to speed up the build process and reduce the size of the image. [performance]
```suggestion
# Add a .dockerignore file with the following content:
# .mvn
# mvnw
# pom.xml
# three-cards-divination/pom.xml
# three-cards-divination/src
# three-cards-divination/target
```